src/aws.js

```diff
@@ -32,3 +33,22 @@ export const clientConfig = {
         httpsAgent: agent,
     }),
 };
+
+/**
+ * Attempt to identify the IAM user name behind the current AWS credentials
+ * (useful for debugging / server start-up messages).
+ */
+export async function getAwsId() {
+  const client = new STSClient({ region: AWS_REGION });
+  const command = new GetCallerIdentityCommand({});
+  try {
+    const data = await client.send(command);
+    if (!data.Arn.startsWith('arn:aws:iam::')) throw new Error("Identity is not an IAM user")
+    return (`IAM user '${data.Arn.replace(/^.+user\//, '')}'`)
+  } catch (error) {
+    if (String(error).includes('Could not load credentials from any providers')) {
```

Comment:
`error` should be an object you can type test against rather than testing against the stringified form which is likely to be more brittle.

Reply:
Yeah, we can get this from `import { CredentialsProviderError } from "@smithy/property-provider";` but I think this is used more widely than the case where credentials aren't provided. I've gone a different way and now just print the strigified error message, which seems to be appropriate from my testing.